src/v/config/configuration.cc

```diff
@@ -1545,7 +1546,46 @@ configuration::configuration()
       "Maximum capacity of rate limit accumulation"
       "in controller configuration operations limit",
       {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
-      std::nullopt) {}
+      std::nullopt)
+  , kafka_throughput_limit_cluster_bps(
+      *this,
+      "kafka_throughput_limit_cluster_bps",
+      "Cluster wide throughput limit - maximum kafka traffic throughput "
+      "allowed through the entire cluster, in bytes/s. Effective cluster wide "
+      "limit may be less due to inertia of balancing this limit between nodes. "
+      "Default is no limit",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
+      std::numeric_limits<
+        decltype(kafka_throughput_limit_cluster_bps)::value_t>::max(),
+      {.min = 1})
```

Comment:
maybe it should an optional type?

Reply:
For the cluster limit, an optional type is on par with this way now in my mind, in terms of pros and cons. I will see how exactly both way will work in balancer when it gets there. For node limit though I think maximum for no-limit is a better way.

Reply:
I've decided in favour of an optional type finally.